packit/schema.py

```diff
@@ -368,10 +368,67 @@ class JobConfigSchema(CommonConfigSchema):
 
     job = EnumField(JobType, required=True, attribute="type")
     trigger = EnumField(JobConfigTriggerType, required=True)
+    # old metadata dictionary keys
+    _targets = TargetsListOrDict(missing=None, data_key="targets")
+    timeout = fields.Integer()
+    owner = fields.String(missing=None)
+    project = fields.String(missing=None)
+    dist_git_branches = fields.List(fields.String(), missing=None)
+    branch = fields.String(missing=None)
+    scratch = fields.Boolean()
+    list_on_homepage = fields.Boolean()
+    preserve_project = fields.Boolean()
+    use_internal_tf = fields.Boolean()
+    additional_packages = fields.List(fields.String(), missing=None)
+    additional_repos = fields.List(fields.String(), missing=None)
+    fmf_url = fields.String(missing=None)
+    fmf_ref = fields.String(missing=None)
+    skip_build = fields.Boolean()
+    env = fields.Dict(keys=fields.String(), missing=None)
+    enable_net = fields.Boolean(missing=True)
+
     metadata = fields.Nested(JobMetadataSchema)
 
+    @pre_load
+    def ordered_preprocess(self, data, **_):
+        if "metadata" in data:
+            logger.warning(
+                "Job metadata keys are deprecated. "
+                "Do not nest any job key under metadata anymore."
+            )
+        for key in ("targets", "dist_git_branches"):
+            if data is dict:
+                if isinstance(data.get(key), str):
+                    # allow key value being specified as string, convert to list
+                    data[key] = [data.pop(key)]
+
+        return data
+
     @post_load
     def make_instance(self, data, **_):
+        if "metadata" in data:
+            metadata = data.pop("metadata")
+            for key in (
+                "_targets",
+                "timeout",
+                "owner",
+                "project",
+                "dist_git_branches",
+                "branch",
+                "scratch",
+                "list_on_homepage",
+                "preserve_project",
+                "use_internal_tf",
+                "additional_packages",
+                "additional_repos",
+                "fmf_url",
+                "fmf_ref",
+                "skip_build",
+                "env",
+                "enable_net",
+            ):
+                if key in metadata:
+                    data[key] = metadata[key]
```

Comment:
So if I understand it correctly, you're trying to convert only the known keys from the metadata dict into the job dict and skip the unknown keys.
That can cause confusion - if a user has for example a typo in one of the metadata keys, that key would be silently skipped and it'd be hard to realize what's a problem without any warning anywhere.
Could we just convert all (probably by just merging the `metadata` dict into the `data` dict?) and let the schema validation do the job?
The warning might be a bit misleading since it'd point to a nonexistent key in a job dict, while the user has the key in the metadata dict, but it's still better than silently skipping it, isn't it?